### PR TITLE
Small UX enhancements

### PR DESCRIPTION
**Pull request by Alexlandeau**
- Renamed "Detected language column" to "Language column"
- Renamed "Categorical column" to "Split by column"
- Added number of words parameter
- Added color palette parameter
- Removed faded color from the palette

### CODE REVIEW

**Comment by Alexlandeau** [comment]
CH62057

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 1]
```diff
@@ -1,115 +1,129 @@
...
+      "type": "COLUMN",
+      "columnRole": "input_dataset",
+      "allowedColumnTypes": ["string"],
+      "label": "Text column",
+      "mandatory": true
+    },
+    {
+      "name": "language",
+      "type": "SELECT",
+      "getChoicesFromPython": true,
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:49]: Forgot to update "Detected language column" -> "Language column" in the dynamic select

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 2]
```diff
@@ -1,115 +1,129 @@
...
+      "type": "COLUMN",
+      "columnRole": "input_dataset",
+      "allowedColumnTypes": ["string"],
+      "label": "Split by column",
+      "description": "Optional column to generate one word cloud per category",
+      "mandatory": false
+    },
+    {
+      "name": "separator_display",
+      "label": "Display",
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:109]:
```suggestion
      "label": "Display parameters",
```

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 3]
```diff
@@ -1,115 +1,129 @@
...
+      "name": "subchart_column",
+      "type": "COLUMN",
+      "columnRole": "input_dataset",
+      "allowedColumnTypes": ["string"],
+      "label": "Split by column",
+      "description": "Optional column to generate one word cloud per category",
+      "mandatory": false
+    },
+    {
+      "name": "separator_display",
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:108]: How about moving the display param section above subchart? To make sure subchart is last

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 4]
```diff
@@ -1,115 +1,129 @@
...
+      "name": "max_words",
+      "label": "Maximum number of words",
+      "minI": 1,
+      "defaultValue": 100
+    },
+    {
+      "type": "STRINGS",
+      "name": "color_list",
+      "label": "Color palette",
+      "description": "A list of hexadecimal color codes",
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:123]:
```suggestion
      "description": "List of hexadecimal color codes or HTML color names to randomly color each word",
```

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 5]
```diff
@@ -1,115 +1,129 @@
...
+      "allowDuplicates": false,
+      "defaultValue": ["#1F75B3", "#FF7F0F", "#2CA02B"]
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:125]: IMO hex codes are a bit frightening for non-technical users. Could you have an option for users to use HTML color names? matplotlib has a function to convert one into the other

**Review by alexcombessie — CHANGES_REQUESTED** [review]

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 6]
```diff
@@ -1,115 +1,129 @@
...
+      "name": "max_words",
+      "label": "Maximum number of words",
+      "minI": 1,
+      "defaultValue": 100
+    },
+    {
+      "type": "STRINGS",
+      "name": "color_list",
+      "label": "Color palette",
+      "description": "A list of colors to randomly color each word. Supported formats are X11 color names, hexadecimal color codes or matplotlib color shortcuts",
```
Comment (alexcombessie) [inline custom-recipes/nlp-visualization-wordcloud/recipe.json:109]: To make it fit on one line:
```suggestion
      "description": "List of colors to randomly color each word. Supported formats are hexadecimal color codes and matplotlib color names.",
```

**Review by alexcombessie — APPROVED** [review]

**Pull request merged**